presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveStorageFormats.java

```diff
@@ -453,22 +453,23 @@ public void testStructTimestamps(StorageFormat format)
                 tableName,
                 format.getStoragePropertiesAsSql()));
 
-        onHive().executeQuery(format(
-                "INSERT INTO %s %s",
-                tableName,
-                TIMESTAMPS_FROM_HIVE.stream()
-                        .map(entry -> format(
-                                "SELECT\n"
-                                        + "%s\n"
-                                        + ",arr(%2$s)\n"
-                                        + ",map(%2$s, %2$s)\n"
-                                        + ",named_struct('col', %2$s)\n"
-                                        + ",array(map(%2$s, named_struct('col', array(%2$s))))\n"
-                                        // some hive versions don't allow insert from bare select
-                                        + "FROM (SELECT 1) t\n",
-                                entry.getId(),
-                                format("TIMESTAMP '%s'", entry.getWriteValue())))
-                        .collect(Collectors.joining(" UNION ALL "))));
+        // Insert in a loop because inserting with UNION ALL sometimes makes values invisible to Presto
```

Comment:
This is concerning, and I don't know what caused it.

Reply:
It is concerning. Did you try to reproduce the problem outside of this PR?

Reply:
I haven't tried to reproduce it outside the PR, but it doesn't seem to depend on anything from the PR (I removed the timestamps and arrays/maps/structs from the test, and it still happened). I'll try to get a working example.

Reply:
It seems like even a simple example works.

```
hive> CREATE TABLE test (col integer);
hive> INSERT INTO test SELECT 1 UNION ALL SELECT 2;
hive> SELECT count(*) FROM test;
2
... (log out of hive, in to Presto)
prestosql> SELECT count(*) FROM hive.default.test;
0
```

Reply:
Is the data visible in Hive, but not in Presto?
or, is it just a bug in INSERT + UNION ALL in (some version(s) of) Hive?

Reply:
Correct. Hive has no problem with the data that I've seen.

Reply:
This sounds like a bug (in Presto) then.
Can you please file an issue (and link it here)?

Reply:
Filed as #6485.

Reply:
Please link to https://github.com/trinodb/trino/issues/6485 in the comment too